corehq/apps/domain/deletion.py

```diff
@@ -112,6 +114,20 @@ def _terminate_subscriptions(domain_name):
         ).update(is_hidden_to_ops=True)
 
 
+def _delete_sql_cases(domain_name):
+    case_accessor = CaseAccessors(domain_name)
+    case_ids = case_accessor.get_case_ids_in_domain()
+    for case_id_chunk in chunked(case_ids, 500):
+        case_accessor.soft_delete_cases(list(case_id_chunk))
```

Comment:
just a thought if there are expected to be a lot of these, it might be worth either giving progress updates that things are still happening or spawning greenlets or something to speed it up.

Reply:
https://github.com/dimagi/commcare-hq/pull/21247/commits/dfe03a6c17bdaa37aad65b2807dadd0f11c2858c